src/api/libcellml/generator.h

```diff
@@ -49,6 +49,12 @@ class LIBCELLML_EXPORT GeneratorVariable
     GeneratorVariable(GeneratorVariable &&rhs) noexcept; /**< Move constructor */
     GeneratorVariable &operator=(GeneratorVariable rhs); /**< Assignment operator */
 
+    template<typename... Args>
+    static std::shared_ptr<GeneratorVariable> create(Args &&... args) noexcept
+    {
+        return std::shared_ptr<GeneratorVariable> {new GeneratorVariable {std::forward<Args>(args)...}};
+    }
+
```

Comment:
Doesn't our best practice require the body of a function to be in a `.cpp` file and not a `.h` file? (Same elsewhere in this PR.)

Reply:
It's a templated method the definition has to be there.

Reply:
Thanks for reminding me why I am not a fan of templates... :)